packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -469,18 +605,36 @@ function FieldControl() {
 		return null;
 	}
 	const titleField = fields.find( ( f ) => f.id === view.titleField );
-	const mediaField = fields.find( ( f ) => f.id === view.mediaField );
+	const previewField = fields.find( ( f ) => f.id === view.mediaField );
 	const descriptionField = fields.find(
 		( f ) => f.id === view.descriptionField
 	);
+
+	const previewFields = fields.filter( ( f ) => f.isPreviewField );
+
+	let previewFieldUI;
+	if ( previewFields.length > 1 ) {
+		const isPreviewFieldVisible =
+			isDefined( previewField ) && ( view.showMedia ?? true );
+		previewFieldUI = (
+			<PreviewFieldItem
+				view={ view }
+				onChangeView={ onChangeView }
+				isVisible={ isPreviewFieldVisible }
+				previewFields={ previewFields }
+				activeField={ previewField }
+			/>
+		);
+	}
 	const lockedFields = [
 		{
 			field: titleField,
 			isVisibleFlag: 'showTitle',
 		},
 		{
-			field: mediaField,
+			field: previewField,
 			isVisibleFlag: 'showMedia',
+			ui: previewFieldUI,
```

Comment:
Is there a way to simplify this implementation? For example, instead of adding a specific UI for a field, what if we made the following changes:

FieldItem:

- add a new property "description" that renders the field in use (e.g.: Featured image).
- add a new property "options" that receives an array with the  available (e.g.: [ "Content preview", "Featured image" ])
- add a new property "onChangeOption" that communicates a new option has been selected

In the `FieldControl`, we do the following for the media field:

```js
<FieldItem
  field={ field }
  description={ view.mediaField.label }
  options=[
    { label:"Content preview", id: content_preview },
    { label: "Featured image", id: feature_image }
  ],
  onChangeOption={ newOption => onChangeView({ mediaField: newOption })
/>
```

I feel the implementation would be simpler if we make the FieldItem (leaf node) a general component that the FieldControl configures depending on the specifics of the field (visible, not visible, can have multiple options, etc.). This way, it's easy to add the same behaviour to other fields (title).

Reply:
I applied this suggestion, It had some complexities but the base idea is there.